src/UI/Implementation/Component/Layout/Page/Renderer.php

```
@@ -0,0 +1,125 @@
+<?php
+
+/* Copyright (c) 2018 Nils Haagen <nils.haagen@concepts.and-training.de> Extended GPL, see docs/LICENSE */
+
+namespace ILIAS\UI\Implementation\Component\Layout\Page;
+
+use ILIAS\UI\Implementation\Render\AbstractComponentRenderer;
+use ILIAS\UI\Renderer as RendererInterface;
+use ILIAS\UI\Component;
+use ILIAS\UI\Implementation\Render\ilTemplateWrapper as UITemplateWrapper;
+use ILIAS\UI\Component\Signal;
+use ILIAS\UI\Component\Image\Image;
+
+class Renderer extends AbstractComponentRenderer {
+	/**
+	 * @inheritdoc
+	 */
+	public function render(Component\Component $component, RendererInterface $default_renderer) {
+		$this->checkComponent($component);
+
+		if ($component instanceof Component\Layout\Page\Standard) {
+			return $this->renderStandardPage($component, $default_renderer);
+		}
```

Please throw a `\LogicException` if we receive an unknown component here as a safeguard and reminder.

---

I thought this was done by $this->checkComponent($component) ?